### ISSUE TITLE
`app-page-layout-tests` failing with timeout

### ISSUE THREAD

**Issue by cfm** [freedomofpress/securedrop#6104]
## Description

Beginning yesterday with `securedrop_ci` run 2984, `app-page-layout-tests` is failing with a timeout in `make validate-test-html`.

## Steps to Reproduce

- https://app.circleci.com/pipelines/github/freedomofpress/securedrop?branch=5828-reply-message
- https://app.circleci.com/pipelines/github/freedomofpress/securedrop?branch=pull%2F6086

## Expected Behavior

`app-page-layout-tests` always passes after `make validate-test-html || true`.

## Actual Behavior

<https://app.circleci.com/pipelines/github/freedomofpress/securedrop/2987/workflows/b2712c98-1396-41a8-9781-430471382e4e/jobs/56712/parallel-runs/1?filterBy=FAILED>:

```
███ Validating HTML source from 72 page-layout test(s)
make: *** [Makefile:231: validate-test-html] Hangup

Too long with no output (exceeded 10m0s): context deadline exceeded
```

<https://app.circleci.com/pipelines/github/freedomofpress/securedrop/2987/workflows/25a47bb5-8bf5-4510-bad1-3edef0e01271/jobs/56716/parallel-runs/1?filterBy=ALL>:

```███ Validating HTML source from 32 page-layout test(s)
make: *** wait: No child processes.  Stop.
make: *** Waiting for unfinished jobs....
make: *** wait: No child processes.  Stop.

Too long with no output (exceeded 10m0s): context deadline exceeded
```

**Comment by conorsch** [comment]
Thanks for jumping on this. At a glance, it seems the breakage is unrelated to any recent changes to CI on our end. Do you have a different understanding? I've just started looking at #6100 and will continue, but will hold off on merge while you debug.

**Comment by cfm** [comment]
@conorsch, that's right; no relevant changes in our configuration since the last successful runs.  What I did not recognize yesterday, but can see plainly today, is that it's `securedrop/bin/dev-shell`, not `html5validator`, that's hanging in:

https://github.com/freedomofpress/securedrop/blob/a6b478f67253accc58f5441dba194a2a8ab50a42/Makefile#L229-L231

I'll see if I can isolate where and why.

**Comment by cfm** [comment]
The culprit addressed in #6105:  `make validate-test-html`'s `dev-shell` was being spawned without the `$DOCKER_BUILD_ARGUMENTS` used by other Docker-based operations in CI, without which it couldn't benefit from Docker's image-layer cache.  This was my omission in #6072.

My best guesses at reconstructing why we haven't seen this failure-mode....

1. **until now:**  `make validate-test-html` was taking *just under* CircleCI's 10-minute timeout, until *some*thing happened to change yesterday (upstream package changes? longer `html5validator` runtime?) to push it over.
2. **in my interactive testing yesterday:**  connecting to the job via SSH disabled CircleCI's timeout on waiting for job-internal output, which let `dev-shell`'s `docker run` invocation finish successfully even without a cache to hit.
